src/UrlHelper.php

```diff
@@ -28,21 +30,17 @@ public function __construct(BaseHelper $helper)
     /**
      * Proxies to `Zend\Expressive\Helper\UrlHelper::generate()`
      *
-     * @param null|string $routeName
-     * @param array $routeParams
-     * @param array $queryParams
-     * @param null|string $fragmentIdentifier
      * @param array $options Can have the following keys:
      *     - router (array): contains options to be passed to the router
      *     - reuse_result_params (bool): indicates if the current RouteResult
      *       parameters will be used, defaults to true
      * @return string
      */
     public function __invoke(
-        $routeName = null,
+        ?string $routeName = null,
```

Comment:
same here, just `string`

Reply:
No; this is necessary due to how the base `UrlHelper` works; you can actually pass `null`, which has the semantic meaning of "use the currently matched route". This is useful if you want to only change a route param or a query param, for instance.

Reply:
you will be able to pass null, because it has default value `null`, it will work and I think I've asked about it once on forum, and it was said, we should go just with `string`. So for exampel if we have:

```php
function a(int $a, string $b = null, ?string $c = null) {}
```

all below calls are correct:
```php
a(1);
a(1, null, 'string');
a(1, 'st', 'ring');
a(2, 'st');
a(2, 'st', null);
...
```

Reply:
Just verified for myself; I assumed that with strict types in place, passing a `null` value here would not work. I'll update momentarily.